### PR TITLE
Reproduce bug

### PR DESCRIPTION
**Pull request by Mykhailenko**
Added test which should reproduce this [bug](https://trello.com/c/3GUVeOLK/2828-stuck-scheduler-10-st-terminate-to-body)

### CODE REVIEW

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+        schedulerHelper = new SchedulerTHelper(false, true);
+        schedulerHelper.createNodeSource("local", 3);
+        schedulerHelper.getSchedulerInterface().changePolicy("functionaltests.rm.PolicyWhichThrowsExceptions");
+        JobId jobId = schedulerHelper.submitJob(new File(runningJob.toURI()).getAbsolutePath());
+        JobId jobId2 = schedulerHelper.submitJob(new File(runningJob1.toURI()).getAbsolutePath());
+        JobId jobId3 = schedulerHelper.submitJob(new File(runningJob2.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventTaskRunning(jobId, "running_task_for20s");
+
+        Thread.sleep(200000); // sleep enough that all jobs should be able to finish
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

**Comment by sonarqube-activeeon** [comment]
SonarQube analysis reported 2 issues
* ![MAJOR][MAJOR] 2 major

Watch the comments in this conversation to review them.

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java [hunk 3]
```diff
@@ -0,0 +1,133 @@
...
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectioWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectionWhileRunning.java [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+        schedulerHelper = new SchedulerTHelper(false, true);
+        schedulerHelper.createNodeSource("local", 3);
+        schedulerHelper.getSchedulerInterface().changePolicy("functionaltests.rm.PolicyWhichThrowsExceptions");
+        JobId jobId = schedulerHelper.submitJob(new File(runningJob.toURI()).getAbsolutePath());
+        JobId jobId2 = schedulerHelper.submitJob(new File(runningJob1.toURI()).getAbsolutePath());
+        JobId jobId3 = schedulerHelper.submitJob(new File(runningJob2.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventTaskRunning(jobId, "running_task_for20s");
+
+        Thread.sleep(200000); // sleep enough that all jobs should be able to finish
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectionWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectionWhileRunning.java [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+        JobId jobId2 = schedulerHelper.submitJob(new File(runningJob1.toURI()).getAbsolutePath());
+        JobId jobId3 = schedulerHelper.submitJob(new File(runningJob2.toURI()).getAbsolutePath());
+
+        schedulerHelper.waitForEventTaskRunning(jobId, "running_task_for20s");
+
+        schedulerHelper.disconnect();
+        schedulerHelper.getSchedulerInterface().disconnect();
+        schedulerHelper.getResourceManager().disconnect();
+
+        Thread.sleep(200000); // sleep enough that all jobs should be able to finish
```
Comment (sonarqube-activeeon) [inline scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectionWhileRunning.java]: MAJOR Remove this use of "Thread.sleep()". [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2925)

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectionWhileRunning.java (hunk 4 above)
Comment (fviale) [reply scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectionWhileRunning.java]: you can use the method waitForEventJobFinished instead

File: scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectionWhileRunning.java (hunk 5 above)
Comment (fviale) [reply scheduler/scheduler-server/src/test/java/functionaltests/rm/TestRMReconnectionWhileRunning.java]: same comment

**Comment by Mykhailenko** [comment]
retest this please

**Pull request merged**